skrub/_data_ops/_utils.py

```diff
@@ -62,3 +66,25 @@ def format_exception(e):
 def format_exception_only(e):
     """compatibility for python < 3.10"""
     return traceback.format_exception_only(type(e), e)
+
+
+def prune_folder(path: str):
+    if not os.path.exists(path):
+        return
+    time_threshold = time.time() - 7 * 24 * 3600  # 7 days ago
+
+    folders = (
```

Comment:
nitpick: the rest of skrub uses the (newer) pathlib module for those kind of filesystem manipulation rather than os, so for consistency it would be great to use it here, especially since the path you get from `_get_output_dir` is already a pathlib.Path 

for example you can get the mtime with [stat](https://docs.python.org/3/library/pathlib.html#pathlib.Path.stat), the Path object representing the directory has iterdir() and glob() methods, you can join with the `/` operator or `joinpath`, etc.